app/models/concerns/resources.rb

```diff
@@ -0,0 +1,8 @@
+module Resources
+  VALID_PASSWORD_REGEX =
+    /\A^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[$&+,:;=?@#|'<>.-^*()%!])[A-Za-z\d$&+,:;=?@#|'<>.-^*()%!]{8,}$\z/
+  VALID_NAME_REGEX = /\A[A-Za-z]+\z/
+  VALID_LOCATION_REGEX = /\A[A-Za-z\s]+\z/
+  VALID_BUSCODE_REGEX = /\A\S*\z/
```

Comment:
i think we should limit the usage of special character here. what do you think? is there any reason to use this regex?

Reply:
At the beginning I tried to use only 5 or 6 special characters. But FactoryBot actually uses all of the special characters that I added in the regex. I was facing errors while testing because of this. That's why I added all the special characters that factoryBot uses here.

Reply:
Bhaia, updated the BUSCODE_REGEX. Thanks for the feedback.